tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoAggregatorMapping.java

```diff
@@ -45,22 +44,31 @@ public class TychoAggregatorMapping extends AbstractTychoMapping {
 
     private static final String TYCHO_POM = "pom.tycho";
 
-    private static final Set<String> COMMON_NAMES = new HashSet<>(
-            Arrays.asList(System.getProperty(TYCHO_POMLESS_AGGREGATOR_NAMES_PROPERTY,
-                    "bundles,plugins,tests,features,sites,products,releng").split(",")));
+    @Override
+    protected String getPackaging() {
+        return "pom";
+    }
+
+    @Override
+    public float getPriority() {
+        return -10f;
+    }
 
     @Override
     protected boolean isValidLocation(String location) {
-        return location.endsWith(TYCHO_POM);
+        return Path.of(location).endsWith(TYCHO_POM);
     }
 
+    private static final ThreadLocal<Boolean> IS_FIRST = ThreadLocal.withInitial(() -> true);
+
     @Override
     protected File getPrimaryArtifact(File dir) {
         File file = new File(dir, TYCHO_POM);
         if (file.exists() && isCurrent(file)) {
             return file;
         }
-        if (COMMON_NAMES.contains(dir.getName().toLowerCase())) {
+        if (IS_FIRST.get().booleanValue()) {
```

Comment:
The mapping methods should **never** be statefull, we can't control, when and how often they are called!

Reply:
OK, then there should be another mean to check that.
I could either create another marker file like 'started-scanning' and check if that is present in the parent or maintain a static ConcurrentHashSet of processed files can can be queried for the same.

Reply:
That you require any marker actually demonstrate that it *does* recurse into sub directories arbitrarily, as said, there should be simply no state in those mappings, and why we limit this to well known names, as otherwise it open the door to strange behaviors.